### PR TITLE
KATI-87_DoctorSettingServer

### PR DESCRIPTION
**Pull request by toannguyen71195**

### CODE REVIEW

File: com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java [hunk 1]
```diff
@@ -634,4 +601,102 @@ public DoctorUserEntity queryDoctorByID(String id) throws SQLException, ClassNot
...
+		try {
+			stmtDelete = con.prepareStatement(deleteQuery.toString());
+			stmtDelete.executeUpdate();
+			stmtInsert = con.prepareStatement(insertQuery.toString());
+			stmtInsert.setInt(1, Integer.parseInt(entity.getId()));
+			for (SpecialityEntity specialityEntity : entity.getSpeciality()) {
+				stmtInsert.setInt(2, Integer.parseInt(specialityEntity.getId()));
+				stmtInsert.executeUpdate();
+			}
+		} finally {
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java]: insert and delete method need rollback if exception happened

**Comment by hieumai** [comment]
OK from me. @vtnlinh95  please start verifying this

**Comment by toannguyen71195** [comment]
[doctor api.txt](https://github.com/hieumai/cura/files/441876/doctor.api.txt)

**Comment by hieumai** [comment]
OK from me. @toannguyen71195 please resolve conflicts and squash so I can merge

**Pull request merged**